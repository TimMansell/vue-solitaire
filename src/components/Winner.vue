<template>
  <div class="winner" data-test="winner">
    <div class="winner__content">
      <h1 class="winner__title">Congratulations, you win!</h1>
      <div class="winner__btn" @click="setGameWon(false)" data-test="winner-btn">
        <NewGame />
      </div>
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex';
import NewGame from '@/components/NewGame.vue';

export default {
  name: 'Winner',
  components: {
    NewGame,
  },
  methods: {
    ...mapActions(['setGameWon']),
  },
};
</script>

<style lang="scss" scoped>
.winner {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  left: 0;
  top: 0;
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.7);
  z-index: 1;

  &__content {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 2rem;
  }

  &__title {
    color: white;
    text-shadow: -1px -1px rgba(0, 0, 0, 0.3);
    text-align: center;
  }
}
</style>
