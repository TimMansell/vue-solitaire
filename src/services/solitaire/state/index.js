const setState = (state, newState) => ({ ...state, ...newState });

export default setState;
