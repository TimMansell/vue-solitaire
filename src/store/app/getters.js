const getters = {
  game: ({ game }) => game,
  isGameLoading: ({ isGameLoading }) => isGameLoading,
  isGamePaused: ({ isGamePaused }) => isGamePaused,
  isOverlayVisible: ({ isOverlayVisible }) => isOverlayVisible,
  gameOutcome: ({ gameOutcome }) => gameOutcome,
  timer: ({ game }) => game.time,
  moves: ({ game }) => game.moves.length,
  hasGameUpdated: ({ hasGameUpdated }) => hasGameUpdated,
  isLatestVersion: ({ isLatestVersion }) => isLatestVersion,
  version: ({ version }) => version,
  isConnecting: ({ connection }) => connection.isConnecting,
  isOnline: ({ connection }) => connection.isOnline,
  isCompletedGame: ({ gameOutcome }) =>
    gameOutcome.hasGameLost || gameOutcome.hasGameWon,
};

export default getters;
