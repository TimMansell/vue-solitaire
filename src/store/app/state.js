// Return state as function so we can reset it.
const state = () => ({
  game: {
    moves: [],
    time: 0,
  },
  gameOutcome: {
    hasGameWon: false,
    hasGameLost: false,
  },
  isGameLoading: true,
  isGamePaused: false,
  isOverlayVisible: true,
  versionMatch: true,
});

export default state;
