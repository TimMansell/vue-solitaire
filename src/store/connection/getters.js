const getters = {
  isConnecting: ({ isConnecting }) => isConnecting,
  hasConnectionError: ({ isConnecting, isOnline }) =>
    !isOnline && !isConnecting,
};

export default getters;
