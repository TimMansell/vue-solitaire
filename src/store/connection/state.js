const state = () => ({
  isConnecting: false,
  hasConnected: false,
  isOnline: false,
});

export default state;
