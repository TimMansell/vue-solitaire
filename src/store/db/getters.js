const getters = {
  userStats: (state) => state.userStats,
  game: (state) => state.game,
};

export default getters;
