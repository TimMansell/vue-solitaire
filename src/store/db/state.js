// Return state as function so we can reset it.
const state = () => ({
  userStats: {
    totalGames: null,
  },
  game: {
    id: null,
    moves: 0,
  },
});

export default state;
