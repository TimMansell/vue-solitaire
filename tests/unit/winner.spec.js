import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Winner from '@/components/Winner.vue';
import state from '@/store/state';
import getters from '@/store/getters';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Card.vue', () => {
  it('matches visible snapshot', () => {
    const wrapper = shallowMount(Winner);

    expect(wrapper).toMatchSnapshot();
  });

  it('should not call store action "setGameWon" when clicked', () => {
    const actions = {
      setGameWon: jest.fn(),
    };

    const store = new Vuex.Store({
      state,
      getters,
      actions,
    });

    const wrapper = shallowMount(Winner, {
      store,
      localVue,
    });

    wrapper.find('[data-test="winner-btn"]').trigger('click');

    expect(actions.setGameWon).toHaveBeenCalled();
  });
});
