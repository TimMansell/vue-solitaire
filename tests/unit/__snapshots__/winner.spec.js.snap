// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue matches visible snapshot 1`] = `
<div data-test="winner" class="winner">
  <div class="winner__content">
    <h1 class="winner__title">Congratulations, you win!</h1>
    <div data-test="winner-btn" class="winner__btn">
      <newgame-stub></newgame-stub>
    </div>
  </div>
</div>
`;
